r/R/dplyr-slice.R

```diff
@@ -0,0 +1,144 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+
+# The following S3 methods are registered on load if dplyr is present
+
+slice_head.arrow_dplyr_query <- function(.data, ..., n, prop) {
+  if (length(group_vars(.data)) > 0) {
+    arrow_not_supported("Slicing Arrow data with groups")
+  }
+
+  if (missing(n)) {
+    n <- prop_to_n(.data, prop)
+  }
+
+  head(.data, n)
+}
+slice_head.Dataset <- slice_head.ArrowTabular <- slice_head.RecordBatchReader <- slice_head.arrow_dplyr_query
+
+slice_tail.arrow_dplyr_query <- function(.data, ..., n, prop) {
```

Comment:
There's an error message for extraneous `slice_*()`-related arguments which dplyr has but we don't have that here:

``` r
library(dplyr)
library(arrow)

mtcars %>%
  slice_tail(n = 3, with_ties = FALSE) %>%
  collect()
#> Error in `slice_tail()`:
#> ! `...` must be empty.
#> ✖ Problematic argument:
#> • with_ties = FALSE

mtcars %>%
  arrow_table() %>%
  slice_tail(n = 3, with_ties = FALSE) %>%
  collect()
#>    mpg cyl disp  hp drat   wt qsec vs am gear carb
#> 1 19.7   6  145 175 3.62 2.77 15.5  0  1    5    6
#> 2 15.0   8  301 335 3.54 3.57 14.6  0  1    5    8
#> 3 21.4   4  121 109 4.11 2.78 18.6  1  1    4    2
```

Reply:
Good catch, added the rlang::check_dots_empty call. I left it as `rlang::` instead of adding it to the importFrom we use because I thought you had some PRs open that were touching that, can move there before merging. (Also need to re-run the docgen.R anyway.)

Reply:
I had a check, and none of my open PRs are touching that, so good to move whenever.